### ISSUE TITLE
Style/ConditionalAssignment  is not disabled 

### ISSUE THREAD

**Issue by Ana06** [rubocop-hq/rubocop#4142]
Style/ConditionalAssignment  not exclude all the offenses when generating `.rubocop_todo.yml`

--------

## Expected behavior

`.rubocop_todo.yml` included  all files with offences for Style/ConditionalAssignment.

## Actual behavior

```
app/controllers/package_controller.rb:23:5: C: Style/ConditionalAssignment: Use the return of the conditional for variable assignment and comparison.
    if (@packages.select{|s| s.project == "#{@default_project}:Update"}.size >0) ...
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/models/appdata.rb:39:5: C: Style/ConditionalAssignment: Use the return of the conditional for variable assignment and comparison.
    if dist == "factory" ...
    ^^^^^^^^^^^^^^^^^^^^
```

## Steps to reproduce the problem

1. Generate `.rubocop_todo.yml` by running `rubocop --auto-gen-config`
2. Jut after Run `rubocop`

And Rubocop complains about the offenses.

## RuboCop version

```
0.47.1 (using Parser 2.4.0.0, running on ruby 2.2.5 x86_64-linux-gnu)
```

**Comment by pocke** [comment]
I can't reproduce the problem.

`test.rb`

```ruby
if cond
  bar
  foo
else
  baz
  foo
end
```


```sh
$ rubocop --auto-gen-config
Inspecting 1 file
C

Offenses:

test.rb:3:3: C: Move foo out of the conditional.
  foo
  ^^^
test.rb:6:3: C: Move foo out of the conditional.
  foo
  ^^^

1 file inspected, 2 offenses detected
Created .rubocop_todo.yml.
Run `rubocop --config .rubocop_todo.yml`, or add `inherit_from: .rubocop_todo.yml` in a .rubocop.yml file.


$ cat .rubocop_todo.yml
# This configuration was generated by
# `rubocop --auto-gen-config`
# on 2017-03-18 17:01:35 +0900 using RuboCop version 0.47.1.
# The point is for the user to remove these configuration records
# one by one as the offenses are removed from the code base.
# Note that changes in the inspected code, or installation of new
# versions of RuboCop, may require this file to be generated again.

# Offense count: 2
Style/IdenticalConditionalBranches:
  Exclude:
    - 'test.rb'


$ rubocop --config .rubocop_todo.yml
Inspecting 1 file
.

1 file inspected, no offenses detected

```

May I need some other configuration?

**Comment by Drenmi** [comment]
@pocke: I'm having this on a project at work with the same cop. Will see if I can isolate a reproducible example next week.

**Comment by Ana06** [comment]
@pocke your error is a different one: `Move foo out of the conditional.`  frm the `Style/IdenticalConditionalBranches` cop, the one I said is `Style/ConditionalAssignment`

To make this cop fail you should use something like:

``` Ruby
if foo
  bar = 1
else
  bar = 2
end
```

**Comment by Drenmi** [comment]
This happened to me again today, with the same files, and another interesting thing happened.

If I have this offense in other files, and list all files in the exclusion list, no more offenses are registered, but when I fixed it in the other files, it reports offenses even though the "broken" files are still in the exclusion list.

I have still not managed to figure out what is special about these files ...

**Comment by Ana06** [comment]
@Drenmi 

> If I have this offense in other files, and list all files in the exclusion list, no more offenses are reported, but when I fix it in the other files, it again reports offenses even though the "broken" files are still in the exclusion list.

Exactly, some offenses are added and some are not :confused:

**Comment by jonas054** [comment]
I'm pretty sure this was solved by #4985. @Ana06 Please open again if the problem is still reproducible on latest gem version or master.
